app/src/main/java/org/simple/clinic/util/RxFoo.kt

```diff
@@ -0,0 +1,14 @@
+package org.simple.clinic.util
+
+import io.reactivex.Maybe
+import io.reactivex.Observable
+import io.reactivex.Single
+import io.reactivex.rxkotlin.Observables
+import org.threeten.bp.Duration
+import java.util.concurrent.TimeUnit
+
+inline fun <reified U> Single<*>.ofType(): Maybe<U> =
+    filter { it is U }
+        .cast(U::class.java)
+
+fun Observables.timer(duration: Duration): Observable<Long> = Observable.timer(duration.toMillis(), TimeUnit.MILLISECONDS)
```

Comment:
I'll use this extension in [FacilityChangeScreenController.kt#L66](https://github.com/simpledotorg/simple-android/blob/27de11b56281f834b5349f3dd7fd3f0a2175676f/app/src/main/java/org/simple/clinic/facility/change/FacilityChangeScreenController.kt#L66) as well once https://github.com/simpledotorg/simple-android/pull/506/ gets merged.